### PR TITLE
Issue#28

### LINKED ISSUE #28: Handle the case that a flashcard with no "question" or no "answer" side exists
## Description
Add a validation for flashcard creation and editing endpoints and a "is answer" field to sides of a flashcard.

## Acceptance Criteria
- [ ] validation added
- [ ] field added

### PR DESCRIPTION
**Pull request by BlueWaves2**
## Description of changes
- cleaned up the schema
- flashcard sides now get validated for mutation operation
- explicit declaration if side is an answer
  
Closes #28 

## How has this been tested?

Please describe the test strategy you followed. If not tested, please explain why.

- [X] automated unit test
- [X] automated integration test
- [ ] manual, exploratory test

    
## Checklist before requesting a review

- [X] My code is easy to understand
- [X] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [X] My code fulfills all acceptance criteria
- [X] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [X] I have made corresponding changes to the documentation or
  the [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [X] I made no breaking changes in the database schema or if so, I will perform a database migration

## Checklist for reviewer

- The code is easy to understand
- The code follows
  the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of
  this project
- The code is tested or if not, the reason is documented or discussed
- The added and existing tests reasonably cover the code change
- The code has no breaking changes in the database schema or if so, the assignee is aware of it and
  will perform a database migration

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/dao/FlashcardSideEntity.java [hunk 1]
```diff
@@ -24,6 +24,9 @@ public class FlashcardSideEntity {
     @Column(nullable = false)
     private boolean isQuestion;
 
+    @Column(nullable = false)
```
Comment (PaulBredl) [inline src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/dao/FlashcardSideEntity.java:27]: are you sure about the nullable false? If so, don't forget the schema migration

**Review by PaulBredl — APPROVED** [review]
lgtm but don't forget the db migration

File: src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/dao/FlashcardSideEntity.java (hunk 1 above)
Comment (BlueWaves2) [reply src/main/java/de/unistuttgart/iste/gits/flashcard_service/persistence/dao/FlashcardSideEntity.java:27]: yeah you're right. It's not really needed

**Review by PaulBredl — CHANGES_REQUESTED** [review]
This is not working in the way you think it is

**Comment by PaulBredl** [comment]
@BlueWaves2 You're not avoiding DB migration this way: "boolean" is a primitive type that cannot have a "null" value. When spring tries to read data from the DB this will lead to a NullPointerException!

And even if you change the type to Boolean, you still need to be careful that no usage of this variable is not nullable, **including the field in the GraphQL schema**.

**Comment by PaulBredl** [comment]
> @BlueWaves2 You're not avoiding DB migration this way: "boolean" is a primitive type that cannot have a "null" value. When spring tries to read data from the DB this will lead to a NullPointerException!
> 
> And even if you change the type to Boolean, you still need to be careful that no usage of this variable is not nullable, **including the field in the GraphQL schema**.

In this case, a DB migration seems easier to do, I would probably rather do that if I were you

**Review by PaulBredl — APPROVED** [review]

**Pull request merged**